File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    d={}
    for p in participant:
        d[p]=d.get(p,0)+1
    
    for c in completion:
        d[c]-=1
    
    for k, v in d.items():
        if v!=0:
            return k
    
    
    
    

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String,Integer> users= new HashMap<>();
        String answer="";
        for(String p:participant){
            users.put(p,users.getOrDefault(p,0)+1);
        }
        for(String c:completion){
            int value=users.get(c);
            value--;
            users.put(c,value);
        }
        for(String p:users.keySet()){
            if(users.get(p)==1){
                answer+=p;
            }
        }
        return answer;
    }
}